### PR TITLE
Data Editor Profiler & Selection Hotfix

### PR DESCRIPTION
**Pull request by stricklandrbls**
- Fixed the non-breaking NaN input detection for the profiler.
- Fixed the TS & Svelte errors / warnings in the profiler.
- Fixed the selection range offsets to match address radix.

### CODE REVIEW

File: src/svelte/src/components/DataMetrics/ByteFrequencyGraph.svelte [hunk 1]
```diff
@@ -138,6 +138,7 @@ limitations under the License.
           const startOffsetTemp = parseInt(e.detail.value)
```
Comment (scholarsmate) [inline src/svelte/src/components/DataMetrics/ByteFrequencyGraph.svelte:138]: the parseInt calls need to be done with respect to the address radix also.  For example `parseInt(e.detail.value, 16)` for hex, etc.

Comment (stricklandrbls) [reply src/svelte/src/components/DataMetrics/ByteFrequencyGraph.svelte:138]: Ah so we're not locking the offset to decimal?

File: src/svelte/src/components/DataMetrics/ByteFrequencyGraph.svelte [hunk 2]
```diff
@@ -324,8 +326,8 @@ limitations under the License.
           >&nbsp;&nbsp;End Offset:
           <Input
             id="end-offset-input"
-            placeholder={endOffset}
-            value={endOffset}
+            placeholder={endOffset.toString()}
+            value={endOffset.toString()}
```
Comment (scholarsmate) [inline src/svelte/src/components/DataMetrics/ByteFrequencyGraph.svelte:330]: Again, toString needs the address radix.

File: src/svelte/src/components/DataMetrics/ByteFrequencyGraph.svelte [hunk 3]
```diff
@@ -352,8 +355,8 @@ limitations under the License.
           >&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Length:
           <Input
             id="length-input"
-            placeholder={length}
-            value={length}
+            placeholder={length.toString()}
+            value={length.toString()}
```
Comment (scholarsmate) [inline src/svelte/src/components/DataMetrics/ByteFrequencyGraph.svelte:359]: I suppose the size ought to use the same radix as the address radix.

File: src/svelte/src/components/DataDisplays/Header/DisplayHeader.svelte [hunk 4]
```diff
@@ -91,9 +92,12 @@ limitations under the License.
...
     start: number,
     end: number,
-    size: number
+    size: number,
+    radix: RadixValues
   ): string {
-    return `${from} [${start} - ${end}] Size: ${size} `
+    return `${from} [${start.toString(radix)} - ${end.toString(
+      radix
+    )}] Size: ${size} `
```
Comment (scholarsmate) [inline src/svelte/src/components/DataDisplays/Header/DisplayHeader.svelte:100]: The size here probably ought to be in the same radix as the address radix.

File: src/svelte/src/components/DataMetrics/ByteFrequencyGraph.svelte (hunk 1 above)
Comment (scholarsmate) [reply src/svelte/src/components/DataMetrics/ByteFrequencyGraph.svelte:138]: Nope.  We'll use whatever radix the address is in.

**Comment by scholarsmate** [comment]
Data offsets and lengths need to be in the same radix as the Address throughout the Data Editor.

**Review by scholarsmate — APPROVED** [review]
+1, 👍

**Review by shanedell — APPROVED** [review]
+1

**Pull request merged**